src/logging.cc

```diff
@@ -1110,7 +1110,7 @@ void LogFileObject::Write(bool force_flush,
        (bytes_since_flush_ >= 1000000) ||
        (CycleClock_Now() >= next_flush_time_) ) {
     FlushUnlocked();
-#ifdef OS_LINUX
+#if defined(OS_LINUX) && !defined(ANDROID)
```

Comment:
newer SDK has posix_fadvise?
http://stackoverflow.com/questions/29347531/undefined-reference-to-posix-fadvise